casetokens.php

```diff
@@ -157,16 +197,18 @@ function casetokens_civicrm_tokens(&$tokens) {
     $tokens['case_roles'] = array(
       'case_roles.client' => ts('Case Client(s)'),
     );
+    $allFields = array_merge(_casetokens_get_contact_fields(), _casetokens_get_contact_custom_fields());
```

Comment:
I think we should use 
```php
$result = civicrm_api3('Contact', 'getfields', [
  'api_action' => "getfields",
]);
```
This API call returns the relevant contact fields and also returns associated custom fields in the format needed. We can also make use of the `title` property of the field to display a user friendly field name rather than the column name of the contact field.

Can you verify this will work as expected?

Reply:
actually this method returns alot of unrelated fields for example api_key, hash,contact_sub_type to name a few so if I opt for this I will have to remove all those unrequired fields